sysl2/sysl/ints_test.go

```diff
@@ -0,0 +1,372 @@
+	// Given
+	args := &intsArg{
+		rootModel: "./tests/",
+		modules:   "integration_excludes_test.sysl",
+		output:    "%(epname).png",
+		project:   "Project",
+	}
+	expectContent := plantumlHeader + `
+[IntegratedSystem] as _0 <<highlight>>
+[System1] as _1 <<highlight>>
+_0 --> _1
+@enduml`
+	expected := map[string]string{
+		"_.png": expectContent,
+	}
+
+	// When
+	result := GenerateIntegrations(args.rootModel, args.title, args.output,
+		args.project, args.filter, args.modules, args.exclude, args.clustered, args.epa)
+
+	// Then
+	assert.Equal(t, expected, result)
+}
+
+func TestGenerateIntegrationsWithPassthrough(t *testing.T) {
+	// Given
+	args := &intsArg{
+		rootModel: "./tests/",
+		modules:   "integration_passthrough_test.sysl",
+		output:    "%(epname).png",
+		project:   "Project",
+	}
+
+	// When
+	result := GenerateIntegrations(args.rootModel, args.title, args.output,
+		args.project, args.filter, args.modules, args.exclude, args.clustered, args.epa)
+
+	// Then
+
+	assert.Len(t, reAs.FindAllString(result["_.png"], -1), 3)
+	assert.Len(t, reAsWithHighlight.FindAllString(result["_.png"], -1), 1)
+	assert.Len(t, rePoint.FindAllString(result["_.png"], -1), 2)
+	assert.Len(t, rePointWithIndirect.FindAllString(result["_.png"], -1), 1)
+}
+
+func TestDoGenerateIntegrations(t *testing.T) {
+	type args struct {
+		flags *flag.FlagSet
+		args  []string
+	}
+	argsData := []string{"ints"}
+	test := struct {
+		name       string
+		args       args
+		wantStdout string
+		wantStderr string
+	}{
+		"Case-Do generate integrations",
+		args{
+			flag.NewFlagSet(argsData[0], flag.PanicOnError),
+			argsData,
+		},
+		"",
+		"",
+	}
+	t.Run(test.name, func(t *testing.T) {
+		stdout := &bytes.Buffer{}
+		stderr := &bytes.Buffer{}
+		DoGenerateIntegrations(test.args.args)
+		assert.Equal(t, test.wantStdout, stdout.String())
+		assert.Equal(t, test.wantStderr, stderr.String())
+	})
+}
+
+func TestHandleStatement(t *testing.T) {
+	intsParam := &collectApplicationDependenciesState{}
+	statement := []*sysl.Statement{
+		{Stmt: &sysl.Statement_Loop{Loop: &sysl.Loop{Stmt: []*sysl.Statement{}}}},
+		{Stmt: &sysl.Statement_LoopN{LoopN: &sysl.LoopN{Stmt: []*sysl.Statement{}}}},
+		{Stmt: &sysl.Statement_Foreach{Foreach: &sysl.Foreach{Stmt: []*sysl.Statement{}}}},
+		{Stmt: &sysl.Statement_Alt{Alt: &sysl.Alt{Choice: []*sysl.Alt_Choice{{Stmt: []*sysl.Statement{}}}}}},
+	}
+
+	intsParam.handleStatement("", "", statement)
+	assert.Len(t, intsParam.deps, 0)
+}
+
+func TestGenerateIntegrationsWithCluster(t *testing.T) {
+	// Given
+	args := &intsArg{
+		rootModel: "./tests/",
+		modules:   "integration_with_cluster.sysl",
+		output:    "%(epname).png",
+		project:   "Project",
+		clustered: true,
+	}
+
+	// When
+	result := GenerateIntegrations(args.rootModel, args.title, args.output,
+		args.project, args.filter, args.modules, args.exclude, args.clustered, args.epa)
+
+	// Then
+	assert.Len(t, reAs.FindAllString(result["_.png"], -1), 3)
```

Comment:
no regex to verify the `package` statement?

Reply:
added